test/suites/alpha/drift/suite_test.go

```diff
@@ -70,13 +71,13 @@ var _ = Describe("Drift", Label("AWS"), func() {
 	var nodeTemplate *v1alpha1.AWSNodeTemplate
 	var provisioner *v1alpha5.Provisioner
 	BeforeEach(func() {
-		customAMI = env.GetCustomAMI("/aws/service/eks/optimized-ami/%s/amazon-linux-2/recommended/image_id", 1)
+		amdAMI = env.GetCustomAMI("/aws/service/eks/optimized-ami/%s/amazon-linux-2/recommended/image_id", 1)
 		nodeTemplate = awstest.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{
 			SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": env.ClusterName},
 			SubnetSelector:        map[string]string{"karpenter.sh/discovery": env.ClusterName},
 		}})
 		provisioner = test.Provisioner(test.ProvisionerOptions{
-			Requirements: []v1.NodeSelectorRequirement{{Key: v1alpha5.LabelCapacityType, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.CapacityTypeOnDemand}}},
+			Requirements: []v1.NodeSelectorRequirement{{Key: v1alpha5.LabelCapacityType, Operator: v1.NodeSelectorOpExists}},
```

Comment:
You still changed the behavior up here. Probably just leave the existing OnDemand behavior and make it an Exists requirement for everywhere that needs it to be more permissive